src/overlays/actors/ovl_En_Guest/z_en_guest.c

```diff
@@ -20,23 +34,200 @@ const ActorInit En_Guest_InitVars = {
     (ActorFunc)EnGuest_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Guest/EnGuest_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Guest/EnGuest_Destroy.s")
+static ColliderCylinderInit_Set3 colliderInit = {
+    { COLTYPE_UNK10, 0x00, 0x00, 0x39, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, 0x00, 0x00, 0x01 },
+    { 10, 60, 0, { 0, 0, 0 } },
+};
+
+static InitChainEntry initChain[] = {
+    ICHAIN_U8(unk_1F, 6, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_4C, 500, ICHAIN_STOP),
+};
+
+UNK_PTR D_80A50BA4[] = {
+    &D_060005FC,
+    &D_060006FC,
+    &D_060007FC,
+};
+
+void EnGuest_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnGuest* this = THIS;
+
+    if ((gSaveContext.infTable[7] & 0x40) != 0) {
+        Actor_Kill(&this->actor);
+    } else {
+        this->objBankIndex = Object_GetIndex(&globalCtx->objectCtx, OBJECT_OS_ANIME);
+        if (this->objBankIndex < 0) {
+            osSyncPrintf(VT_COL(RED, WHITE));
+            osSyncPrintf("%s[%d] : バンクが無いよ！！\n", "../z_en_guest.c", 129); // No such bank!!
+            osSyncPrintf(VT_RST);
+            __assert("0", "../z_en_guest.c", 132);
+        }
+    }
+}
+
+void EnGuest_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnGuest* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->colliderCylinder);
+}
+
+void EnGuest_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnGuest* this = THIS;
+    u32 padding;
+
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->objBankIndex) != 0) {
+        this->actor.flags &= ~0x10;
+        Actor_ProcessInitChain(&this->actor, initChain);
+
+        SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_060000F0, NULL, this->limbDrawTable, this->transitionDrawTable,
+                         0x10);
```

Comment:
Limb count is generally better as decimal.
```suggestion
        SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_060000F0, NULL, this->limbDrawTable, this->transitionDrawTable,
                         16);
```